src/main/java/athena/Athena.java

```diff
@@ -29,17 +49,30 @@ public static void main(String[] args) {
         storage.saveTaskListData(taskList);
         command = parser.parse("delete 2");
         command.execute(taskList, athena.ui);
-        storage.saveTaskListData(taskList);
+        storage.saveTaskListData(taskList);*/
 
     }
 
     public void runProgram() {
+        Scanner input;
+        String inputString;
+        Command userCommand;
+
+        ui.printWelcomeMessage();
 
+        oldTask = storage.loadTaskListData();
+
+        while (true) {
+            try {
+                input = new Scanner(System.in);
```

Comment:
We shouldn't instantiate the `Scanner` in every iteration of the loop. From experience this causes problems with the IO redirection tests.